android-glue/src/lib.rs

```diff
@@ -0,0 +1,278 @@
+use android_ndk::input_queue::InputQueue;
+use android_ndk::looper::ThreadLooper;
+use android_ndk::native_activity::NativeActivity;
+use android_ndk::native_window::NativeWindow;
+use android_ndk_sys::{AInputQueue, ANativeActivity, ANativeWindow, ARect, ALOOPER_EVENT_INPUT};
+use lazy_static::lazy_static;
+use log::Level;
+use std::ffi::{CStr, CString};
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+use std::os::raw;
+use std::os::unix::prelude::*;
+use std::ptr::NonNull;
+use std::sync::RwLock;
+use std::thread;
+
+pub fn android_log(level: Level, tag: &CStr, msg: &CStr) {
+    use android_log_sys::LogPriority;
+    let prio = match level {
+        Level::Error => LogPriority::ERROR,
+        Level::Warn => LogPriority::WARN,
+        Level::Info => LogPriority::INFO,
+        Level::Debug => LogPriority::DEBUG,
+        Level::Trace => LogPriority::VERBOSE,
+    };
+    unsafe {
+        android_log_sys::__android_log_write(
+            prio as _,
+            tag.as_ptr() as *const _,
+            msg.as_ptr() as *const _,
+        );
+    }
+}
+
+lazy_static! {
+    static ref NATIVE_WINDOW: RwLock<Option<NativeWindow>> = Default::default();
+    static ref INPUT_QUEUE: RwLock<Option<InputQueue>> = Default::default();
+    static ref CONTENT_RECT: RwLock<Rect> = Default::default();
```

Comment:
What the reason you select RwLock rather than Mutex?

Reply:
Actually I don't sure what is the best choise in this case.

Reply:
It's a `RwLock` because the input thread is the only one that should write, while multiple consumers can read and block the input thread concurrently. If it's a mutex there is no way of preventing consumers from writing to it.

Reply:
Ok, I understand.